### ISSUE TITLE
Keep same database name when changing data source

### ISSUE THREAD

**Issue by webian** [dbeaver/dbeaver#20338]
**Is your feature request related to a problem? Please describe.**
I often have to run a query on my dev server and then on my production server or vice versa.
Unfortunately, at the moment, when I change source from the "Select data source" menu, the selected database changes to the one that is first in the list which often does not match to the one that was previously selected. So I have to reselect the database with the same name but in the other server.

**Describe the solution you'd like**
I would like that when I switch servers the database with the same name in the other server would be selected, if present.
If it is not present, I would like no database to be selected, so that I have to manually choose the right one (this is to avoid accidentally running a query in the first database when it is not the one I want).

**Comment by ShadelessFox** [comment]
Hello @webian,

Have you tried changing the default database/schema names for your database connections?

**Comment by webian** [comment]
My feature request is not about default DB, it's about NOT changing the current DB name when changing server.

**Comment by webian** [comment]
Having:

- Server1
  - Database1
  - Database2
  - Database3
- Server2
  - Database1
  - Database2
  - Database3

Steps:
1. I'm in Server1+Database2.
2. I switch to Server2.
3. Now the selected DB is Database1 not the same database before switch: Database2

**Comment by ShadelessFox** [comment]
Yes, because DBeaver changes the active database and schema to defaults of the selected connection.

I think preserving previously selected database/schema and ignoring defaults is a bad idea, both conceptually and in terms of backward compatibility.

**Comment by webian** [comment]
Could you please elaborate on why you think it's a bad idea?

**Comment by ShadelessFox** [comment]
Because defaults exist for a reason, they are used when creating a new script or opening a console. It seems intuitive to do the same when changing data source association for a file.

This workaround is unhelpful if you're operating over several databases/schemas.

**Comment by webian** [comment]
My feature request is an **option** that sets this behaviour in DBeaver:
"If this option is set, when I switch Data Source, try to keep as active the database with the same name in the now active Data Source"

I find this option very useful in my use case because I often have to switch between development Data Source and production Data Source. Keep in mind that I have many DBs on dev Data Source and pretty much the same DBs in production Data Source.

Sorry if I sound stubborn but in my case having to reselect the database I'm working on every time I switch from development server to production server is really annoying.

**Comment by ShadelessFox** [comment]
Adding an option sounds nice. We can agree on that proposal. Thanks for the clarification on your request.

**Comment by webian** [comment]
Another small improvement related to this feature request is to set the focus to the filter input field of the "Choose catalog/schema" popup like it happens for the "Select Data Source" popup:

![uhxi2bkj](https://github.com/dbeaver/dbeaver/assets/462601/4f3971d4-1745-437d-9f14-760a3f03b502)

so one can CTRL+D and start typing the DB name to filter the list.
